### PR TITLE
improve `Index[View].getKnownUsers()`

### PR DESCRIPTION
**Pull request by Ladicek**
See also https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/hibernate.20reactive/near/424410591

### CODE REVIEW

File: core/src/main/java/org/jboss/jandex/Indexer.java [hunk 1]
```diff
@@ -2557,7 +2622,11 @@ public Index complete() {
         propagateTypeVariables();
 
         try {
-            return new Index(masterAnnotations, subclasses, subinterfaces, implementors, classes, modules, users);
+            Map<DotName, List<ClassInfo>> userLists = new HashMap<>();
+            for (Map.Entry<DotName, Set<ClassInfo>> entry : users.entrySet()) {
+                userLists.put(entry.getKey(), new ArrayList<>(entry.getValue()));
```
Comment (mkouba) [inline core/src/main/java/org/jboss/jandex/Indexer.java:2627]: I wonder if `List#copyOf()` could be a little bit more efficient here?

Comment (Ladicek) [reply core/src/main/java/org/jboss/jandex/Indexer.java:2627]: Possibly, but Jandex is still on Java 8 :-)

Comment (mkouba) [reply core/src/main/java/org/jboss/jandex/Indexer.java:2627]: Ah, sorry. I forgot about it :facepalm:

**Review by mkouba — APPROVED** [review]
Looks good but I'm not 100% sure it's a good idea to target a bugfix release...

**Comment by Ladicek** [comment]
I often add miniature features in micro releases, but you're probably right -- this should go into 3.2.0.

I'll change the milestone and make this a draft for now.

**Review by FroMage — APPROVED** [review]
I suppose this is more general, and more useful than just appearing in the constant pool, although I suppose that in the first two use-cases we've had (calling static methods on `Log` and calling any method/field on Panache entities) it's now going to be possible to get "users" that "only" declare our types (`Log` or Panache entities) somewhere in their signatures without calling any method on them, which may be "false positives" in these cases.

In the case of Panache, we'll register a class visitor/transformer too many, but it won't do anything since it won't find any field access to transform. The chances of classes having signature references to Panache entities without using their fields is plausible.

In the case of Quarkus `Log` it may be that we define an extra static logger for a class that will go unused, but the chances of that happening is pretty unlikely.

**Comment by Ladicek** [comment]
Agree. Thanks!

**Pull request merged**